src/main/java/gim/logic/commands/GenerateCommand.java

```diff
@@ -28,47 +29,65 @@ public class GenerateCommand extends Command {
     public static final String COMMAND_WORD = ":gen";
     public static final String DIFFICULTY_LEVELS = "{easy, medium, hard}";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD
-            + " -> Generates a sample workout based on PRs (Personal Records) "
-            + "of the exercise(s) identified by the indices number used in the displayed exercise list, "
-            + "according to the difficulty level specified. " + "\n"
-            + "Difficulty levels supported are: " + DIFFICULTY_LEVELS + "\n"
-            + "Parameters: INDEX_1 [, INDEX_2, ...] (each index must be a positive integer) "
-            + PREFIX_LEVEL + "LEVEL (must be one of " + DIFFICULTY_LEVELS + ") " + "\n"
-            + "Example usage: " + COMMAND_WORD + " 2,3 " + PREFIX_LEVEL + "easy";
+    public static final String MESSAGE_USAGE = "The generate feature has two variations: \n\n"
+            + COMMAND_WORD + " INDEX [, INDEX]... " + PREFIX_LEVEL + "LEVEL\n"
+            + "Each INDEX must be a positive integer; LEVEL must be one of " + DIFFICULTY_LEVELS + ".\n"
+            + "Example usage: \n" + COMMAND_WORD + " 2,3 " + PREFIX_LEVEL + "easy\n\n"
+            + COMMAND_WORD + " n/NAME [n/NAME]... level/LEVEL\n"
+            + "NAME must be of an exercise found in the system; LEVEL must be one of " + DIFFICULTY_LEVELS + ".\n"
+            + "Example usage: \n" + COMMAND_WORD + " n/squat n/deadlift " + PREFIX_LEVEL + "easy";
 
     public static final String MESSAGE_GENERATE_SUCCESS = " workout session generated: \n";
 
-
     private final ArrayList<Index> indices;
     private final ValidLevel level;
+    private final Set<Name> nameSet;
 
 
     /**
      * @param indices of the exercises in the filtered exercise list.
-     * @param level difficulty level of the workout generated.
+     * @param level   difficulty level of the workout generated.
      */
     public GenerateCommand(ArrayList<Index> indices, ValidLevel level) {
         requireAllNonNull(indices, level);
         this.indices = indices;
         this.level = level;
+        this.nameSet = null;
+    }
+
+    /**
+     * @param nameSet containing names of the exercises given by user, not verified as valid.
+     * @param level   difficulty level of the workout generated.
```

Comment:
This one as well, is this whitespace supposed to be here?

Reply:
resolved!